modules/dnn/src/caffe/caffe_importer_v2.cpp

```diff
@@ -0,0 +1,579 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2017, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+#ifndef HAVE_PROTOBUF
+#include <typeinfo>
+
+#include "../precomp.hpp"
```

Comment:
`precomp.hpp` must be the first include without any `#if`s

Reply:
Fixed